contracts/claimManager.sol

```diff
@@ -0,0 +1,242 @@
+pragma solidity ^0.4.0;
+
+import {DepositsManager} from './DepositsManager.sol';
+import {ScryptVerifier} from "./scryptVerifier.sol";
+
+// ClaimManager: queues a sequence of challengers to play with a claimant.
+
+contract ClaimManager is DepositsManager {
+    uint private numClaims = 0;     // index as key for the claims mapping.
+    uint public minDeposit = 1;    // TODO: what should the minimum deposit be?
+
+    // a claim goes through two stages after being created:
+    // stage 1: new challenges are allowed.
+    // stage 2: no new challenges are allowed.
+    // the claim exists until all verification games are played out.
+    // reasoning behind having two stages is to prevent DoS attacks;
+    // i.e. a malicious adversary cannot sybil challenge a claim after challengeTimeout blocks have passed.
+    uint public challengeTimeout = 20;
+
+    // blocks to wait for callback from scryptVerifier,
+    // before moving onto the next.
+    // TODO: we're not using this yet.
+    uint public verificationGameTimeout = 40;
+
+    address public dogeRelayAddress;
+    address public scryptVerifierAddress;
+
+    event DepositBonded(uint claimID, address account, uint amount);
+    event DepositUnbonded(uint claimID, address account, uint amount);
+    event ClaimCreated(uint claimID, address claimant, bytes plaintext, bytes blockHash);
+    event ClaimChallenged(uint claimID, address challenger);
+    event ClaimVerificationGameStarted(uint claimID, address claimant, address challenger, uint sessionId);
+    event ClaimDecided(uint claimID, address winner, address loser);
+    event ClaimSuccessful(uint claimID, address claimant, bytes plaintext, bytes blockHash);
+    event ClaimVerificationGamesEnded(uint claimID);
+
+    struct ScryptClaim {
+        uint id;
+        address claimant;
+        bytes plaintext;    // the plaintext Dogecoin block header.
+        bytes blockHash;    // the Dogecoin blockhash.
+        uint createdAt;     // the block number at which the claim was created.
+        mapping (uint => address) challengers;      // all current challengers.
+        uint numChallengers;
+        uint currentChallenger;    // index of next challenger to play a verification game.
+        bool verificationOngoing;   // is the claim waiting for results from an ongoing verificationg game.
+        mapping (address => uint) bondedDeposits;   // all deposits bonded in this claim.
+    }
+
+    mapping(uint => ScryptClaim) private claims;
+
+    modifier onlyBy(address _account) {
+        require(msg.sender == _account);
+        _;
+    }
+
+    ScryptVerifier sv;
+
+    // @dev – the constructor
+    function ClaimManager(address _dogeRelayAddress, address _scryptVerifierAddress) public {
```

Comment:
This should be an interface and an address.

Reply:
Noted in issue https://github.com/TrueBitFoundation/scrypt-interactive/issues/22.